gateway-js/src/config.ts

```diff
@@ -87,13 +133,39 @@ export interface RemoteGatewayConfig extends GatewayConfigBase {
   introspectionHeaders?: HeadersInit;
 }
 
-export interface ManagedGatewayConfig extends GatewayConfigBase {
+// TODO(trevor:cloudconfig): This type goes away
+export interface LegacyManagedGatewayConfig extends GatewayConfigBase {
   federationVersion?: number;
 }
 
-interface ManuallyManagedGatewayConfig extends GatewayConfigBase {
+// TODO(trevor:cloudconfig): This type becomes the only managed config
+export interface PrecomposedManagedGatewayConfig extends GatewayConfigBase {
+  /**
+   * @deprecated This configuration option shouldn't be used unless by
+   *             recommendation from Apollo staff. This behavior will be
+   *             defaulted in a future release and this option will strictly be
+   *             used as an override.
+   */
+  experimental_schemaConfigDeliveryEndpoint: string;
```

Comment:
Tying in with my comment that options should override env vars, this should be `| null` where an explicit null means to ignore an env var (which will imply a slight change to isPrecomposedManagedConfig).

Reply:
fcf5fc4